### DISCUSSION TITLE
Please align Cost Export functionality for different subscription models (CSP specifically)

### DISCUSSION THREAD

**Discussion by mlohmeijer** [microsoft/finops-toolkit#1204] [Ideas]
Right now the data model, ability to export and datasets are different for different subscription models. Specifically the capabilities to get cost exports for CSP subscriptions are not what they should be to support a FinOps practitioner working in-house for a Microsoft end-customer or independent FinOps practitioner providing FinOps capabilities to Microsoft CSP customers. 

Examples:
- The CSP has to allow cost visibility explicitly. If they don't then you have an issue and moving CSPs is not like buying your coffee from a different café. This should be open for any customer directly. 
- If the CSP does not allow cost visibility then you only have the monthly reconciliation file to fall back on (if you can get it from the CSP) and it lacks all detail. This also removes the ability to correlate cost and usage with utilization (outside of what Azure Advisor gives). 
- Each export has to be made on subscription level and not on the tenant root group. We're working on some automation here and will contribute if we get something halfway decent (provided the API can spit out a list of exports per subscription, otherwise you would have to recursively check each subscriptionId for a corresponding export) but this should be standard functionality. 
- The CSP export data model is different from the EA data model. You could switch to FOCUS format but the export hasn't been updated to v1.1 yet. 

Microsoft is actively promoting customers to move to CSP and by the time these customers come around to look at FinOps it's no longer an option to discuss whether they shouldn't move to MCA-E. The available tools for managing FinOps for CSP subscriptions should be updated accordingly.

**Comment by flanakin** [comment]
Sorry to hear you're facing these challenges. Unfortunately, these are constraints in Cost Management and not something we can work around from the FinOps toolkit without working directly with your partner and building some specialized capabilities exclusively for CSP partners. That said, I'm totally open to partnering with anyone who's able to contribute in this area.

Please note that Cost Management has not added FOCUS 1.1 support for any account type. It's not a CSP limitation. That said, this is one area we can work on. We are hoping to add FOCUS 1.1 support in the next few months. You can vote that up here: https://github.com/microsoft/finops-toolkit/issues/1275
